File: NeetCode/javascript/Hashing/0349. Intersection of Two Arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function(nums1, nums2) {
    const res=[]
    const s1=new Set(nums1);
    const s2=new Set(nums2);
   for(num of s1)
   {
       if(s2.has(num))
       {
           res.push(num)
       }
   }
   return res;
};

File: NeetCode/C++/Hashing/0349. Intersection of Two Arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int>s{nums1.begin(),nums1.end()};
        vector<int>v;
        for(auto itr=nums2.begin();itr!=nums2.end();itr++)
        {
            if(s.find(*itr)!=s.end())
            {
                v.push_back(*itr);
                s.erase(*itr);
            }
        }
        return v;
    }
};